flatfs.go

```diff
@@ -606,6 +624,12 @@ func (fs *Datastore) putMany(data map[datastore.Key][]byte) error {
 		if _, err := tmp.Write(value); err != nil {
 			return err
 		}
+
+		// TODO: I don't understand why the temp files are not closed immediately
+		//  here. After this point we either rename or delete them so why wait?
+		//  It induces the complexity of the closer to avoid having too much open
+		//  files.
+		// TODO: concurrent writes?
```

Comment:
> Closing the tmp files will result in the data being synced to disk

Ha right, that's a very good point. So that way it's up to the kernel to stream that data on disk however makes the more sense.